docs/dev/guides/bootstrap.md

```diff
@@ -0,0 +1,196 @@
+
+    ID_PREFIX = "my_resource"
+
+    def example_action(self, data: t.Dict[str, t.Any]) -> t.Dict[str, t.Any]:
+        """Create a custom resource with specific data.
+
+        Args:
+            data: The data dictionary to send to the API
+
+        Returns:
+            The created resource entity
+        """
+        response = self.call(
+            operation="example_action",
+            body=data,
+        )
+        return response
+```
+
+### Modify Command Group
+
+Edit `pulpcore/cli/my_plugin/__init__.py` to define your command groups and add functionality:
+
+```python
+import typing as t
+import click
+from pulp_cli.generic import pulp_group
+from pulp_glue.common.i18n import get_translation
+
+# Import your command plugins
+from pulpcore.cli.my_plugin.my_command import my_command_group
+
+translation = get_translation(__package__)
+_ = translation.gettext
+
+__version__ = "0.1.0"  # Matches version in pyproject.toml
+
+
+@pulp_group(name="my_plugin")
+def my_plugin_group() -> None:
+    """My plugin commands."""
+    pass
+
+
+def mount(main: click.Group, **kwargs: t.Any) -> None:
+    # Add your command groups here
+    my_plugin_group.add_command(my_command_group)
+    main.add_command(my_plugin_group)
+```
+
+### Create Command plugins
+
+Create a new file for your commands, e.g., `pulpcore/cli/my_plugin/my_command.py`:
+
+```python
+import click
+from pulp_cli.common.context import pass_pulp_context
+from pulp_glue.common.context import PulpContext
+from pulp_glue.my_plugin.context import PulpMyExampleContext
+from pulp_cli.common.generic import pass_entity_context
+
+
+@click.group()
+@pass_pulp_context
+@click.pass_context 
+def my_command_group(ctx: click.Context, pulp_ctx: PulpContext, /) -> None:
+    """My custom commands."""
+    ctx.obj = PulpMyExampleContext(pulp_ctx)
+
+
+@my_command_group.command()
+@click.option("--data", required=True, help="Data for the example action")
+@pass_entity_context
+@pass_pulp_context
+def create(pulp_ctx: PulpContext, my_example_ctx: PulpMyExampleContext, /, data: str):
+    """Calling example action."""
+    result = my_example_ctx.example_action({"data": data})
+    pulp_ctx.output_result(result)
+```
+
+---
+
+## Development Workflow
+
+### Install Your plugin in Development Mode
+
+For installation instructions, see here [installation.md].
+
+### Test Your Commands
+
+After installation, you can test your commands:
+
+```bash
+pulp my-plugin my-command create --data "example"
+```
+
+Consider writing tests for your commands as soon as you implement them.
+[testing doc]
+
+---
+
+## Update Templates
+
+If you need to update the project structure with new template changes:
+
+```bash
+cd pulp-cli-my-plugin
+../pulp-cli/cookiecutter/apply_templates.py
+```
+
+---
+
+By using the bootstrap process, you can set up a properly structured Pulp CLI plugin and focus on implementing your custom commands rather than worrying about project structure.
```

Comment:
I wonder what the intention of this paragraph is. When someone made it to here, I hope they knew what they'd been doing.
It sounds more like a motivational thing that can go in the frontmatter. But there is something similar too.
Can we merge them somehow?
Or as being a guide it should read like " At this point you have accomplished ..., next steps are...".